ebean-datasource/src/main/java/io/ebean/datasource/pool/ConnectionPool.java

```diff
@@ -725,6 +735,77 @@ private void stopHeartBeatIfRunning() {
     }
   }
 
+  static class AsyncCloser implements Runnable {
+    final PooledConnection pc;
+    final boolean logErrors;
+
+    public AsyncCloser(PooledConnection pc, boolean logErrors) {
+      this.pc = pc;
+      this.logErrors = logErrors;
+    }
+
+    @Override
+    public void run() {
+      pc.doCloseConnection(logErrors);
+    }
+
+    @Override
+    public String toString() {
+      return pc.toString();
+    }
+  }
+
+  /**
+   * Tries to close the pc in an async thread. The method waits up to 5 seconds and returns true,
```

Comment:
javadoc isn't correct here. The executor/thread doing the closing will just block imo.